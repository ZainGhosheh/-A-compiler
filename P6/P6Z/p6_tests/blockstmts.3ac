[BEGIN GLOBALS]
main
str_1 "filler command"
str_0 "two ehs"
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN main LOCALS]
a (local var of 8 bytes)
a (local var of 8 bytes)
varTmp0 (tmp var of 8 bytes)
varTmp1 (tmp var of 8 bytes)
varTmp2 (tmp var of 8 bytes)
varTmp3 (tmp var of 8 bytes)
varTmp4 (tmp var of 8 bytes)
varTmp5 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            call randBool
            getret [varTmp0]
            call randBool
            getret [varTmp1]
            [varTmp2] := [varTmp0] AND64 [varTmp1]
            IFZ [varTmp2] GOTO lbl_2
            WRITE str_0
lbl_2:      nop
            [varTmp3] := 1 ADD64 2
            [varTmp4] := [varTmp3] GT64 3
            IFZ [varTmp4] GOTO lbl_3
            [varTmp5] := 4 DIV64 6
            [a] := [varTmp5]
            goto lbl_4
lbl_3:      nop
            [a] := 10
lbl_4:      nop
lbl_5:      nop
            IFZ 1 GOTO lbl_6
            WRITE str_1
            goto lbl_5
lbl_6:      nop
            setret 2
            goto lbl_1
lbl_1:      leave main

