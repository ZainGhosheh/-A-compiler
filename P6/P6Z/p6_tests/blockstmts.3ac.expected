[BEGIN GLOBALS]
main
str_1 "filler command"
str_0 "two ehs"
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN main LOCALS]
a (local var of 8 bytes)
a (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
tmp2 (tmp var of 8 bytes)
tmp3 (tmp var of 8 bytes)
tmp4 (tmp var of 8 bytes)
tmp5 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            call randBool
            getret [tmp0]
            call randBool
            getret [tmp1]
            [tmp2] := [tmp0] AND64 [tmp1]
            IFZ [tmp2] GOTO lbl_2
            WRITE str_0
lbl_2:      nop
            [tmp3] := 1 ADD64 2
            [tmp4] := [tmp3] GT64 3
            IFZ [tmp4] GOTO lbl_3
            [tmp5] := 4 DIV64 6
            [a] := [tmp5]
            goto lbl_4
lbl_3:      nop
            [a] := 10
lbl_4:      nop
lbl_5:      nop
            IFZ 1 GOTO lbl_6
            WRITE str_1
            goto lbl_5
lbl_6:      nop
            setret 2
            goto lbl_1
lbl_1:      leave main